ch12.md

```diff
@@ -0,0 +1,256 @@
+
+An interface without laws is merely indirection. Like any other mathematical structure, we must expose properties for our own sanity. This has a similar effect as encapsulation since it protects the data, enabling us to swap out the interface with another law abiding citizen.
+
+Come along now, we've got some laws to suss out.
+
+### Identity
+
+```js
+const identity1 = compose(sequence(Identity.of), map(Identity.of))
+const identity2 = Identity.of
+
+// test it out with Right
+identity1(Right('stuff'))
+// Identity(Right('stuff'))
+
+identity2(Right('stuff'))
+// Identity(Right('stuff'))
+```
+
+This should be straight forward. If we place an `Identity` in our functor, then turn it inside out with `sequence` that's the same as just placing it on the outside to begin with. We chose `Right` as our guinea pig as it is easy to try the law and inspect. An arbitrary functor there is normal, however, the use of a concrete functor here, namely `Identity` in the law itself might raise some eyebrows. Remember a [category](ch5.md#category-theory) is defined by morphisms between its objects that have associative composition and identity. When dealing with the category of functors, natural transformations are the morphisms and `Identity` is, well identity. The `Identity` functor is as fundamental in demonstrating laws as our `compose` function. In fact, we should give up the ghost and follow suit with our [Compose](ch8.md#a-spot-of-theory) type:
+
+### Composition
+
+```js
+const comp1 = compose(sequence(Compose.of), map(Compose.of))
+const comp2 = (point1, point2) => compose(Compose.of, map(sequence(point2)), sequence(point1))
+
+
+// Test it out with some types we have lying around
+comp1(Identity(Right([true])))
+// Compose(Right([Id(true)]))
+
+comp2(Right, Array)(Identity(Right([true])))
+// Compose(Right([Id(true)]))
+```
+
+This law preserves composition as one would expect: if we swap compositions of functors, we shouldn't see any surprises since the composition is a functor itself. We arbitrarily chose `true`, `Right`, `Identity`, and `Array` to test it out. Libraries like [quickcheck](https://hackage.haskell.org/package/QuickCheck) or [jsverify](http://jsverify.github.io/) can help us test the law by fuzz testing the inputs.
+
+As a natural consequence of the above law, we get the ability to fuse traversals:
+
+```js
+traverse(compose(Compose.of, map(compose(f , g)))) = compose(Compose.of, map(traverse(of, f)), traverse(of, g))
+```
+
+Which is nice from a performance standpoint.
+
+### Naturality
+
+```js
+const natLaw1 = (point, nt) => compose(nt, sequence(p))
+const natLaw2 = (point, nt) => compose(sequence(p), map(nt))
```

Comment:
Same consideration from above about using `f` instead of `point` (or `p` with the typo ^.^).